src/file_storage.cpp

```diff
@@ -802,7 +802,7 @@ namespace libtorrent
 	std::int64_t file_storage::file_size(int const index) const
 	{
 		TORRENT_ASSERT_PRECOND(index >= 0 && index < int(m_files.size()));
-		return std::int64_t(m_files[std::size_t(index)].size);
+		return m_files[std::size_t(index)].size;
```

Comment:
is it possible to disable the specific warning that causes this? or would that disable too many cases?